### ISSUE TITLE
How do I find Coltan Bedrock Ore? (528 Mode and Ore Config are enabled, Coltass points to middle of nowhere)

### ISSUE THREAD

**Issue by GregoriusT** [HbmMods/Hbm-s-Nuclear-Tech-GIT#281]
I recently got a Message that you added a Coltan Bedrock Ore, so I was curious how yours worked and specifically how it spawned. Sadly, after updating and reading changelog ofcourse, even with the Coltass, and with all 528 Configs enabled, I was not able to find Coltan.

Apart from the sporadic ones, which I DID find after enabling them too. I even generated a completely fresh world in case deleting region Files was not the way to regenerate it.

So the Coltan Mother Lode seems to not be working yet, I presume?

Hard to make things work with my own Bedrock Drill, if I don't know how many Coltan Bedrock Ore spots exist on average.

P.S. It is spelled Deposit*, not "Depsoit" how it is in parts of the Config File (obviously a typo). ;)

**Comment by HbmMods** [comment]
The coltan deposit was actually the first coltan related ore gen that was implemented and confirmed to work, I found the expected amounts after making a new world just now and it's also confirmed to work on a public server. It is however possible that the compass points in the wrong direction in the unlikely case that the world's seed changes somehow in between the world gen events and the item updates. The compass will also refuse to work at all if Optifine is installed with item animations turned off.

The coltan _bedrock_ deposits however don't work yet due to a bug I found yesterday. They are expected to spawn once every 50 chunks by default, and yield one slab of coltan (half an ingot of tantalum after processing) with an automatic mining drill around every 10ish seconds (assuming a speed III upgrade).

**Comment by GregoriusT** [comment]
> It is however possible that the compass points in the wrong direction in the unlikely case that the world's seed changes somehow in between the world gen events and the item updates

Wait how the heck would THAT happen?! Okay I'm gonna look at the Code to see if there is something obvious...

...So I looked at the Coltass Code and it seems to set NBT in order to sync the result over to the Client, which seems to be fine...

...Ah I see, Coltan generates at a HUGE range of 750 Blocks, so what i ASSUMED was the normally generating Coltan was clearly the actual large deposit (I expected a bit more density, or depth-ore-alikes XD). So everything is actually fine, except for the aforementioned Bedrock ones ofcourse.

And if Bedrock Coltan really supposed to spawn in the middle of World Height? I thought it would have made more sense for it to be part of the Bedrock Floor itself.

Well, if it is a "one Block of Bedrock Ore"-System, I know how to balance it with my own Bedrock Ores, since mine usually do a 6x6 of randomly sprayed Bedrock Ores, where you have to select a 3x3 to place the Bedrock Drill ontop of, in order to maximize yield. Or invest into four Bedrock Drills for the whole thing. ^^

Bedrock Ores and huge Ore Veins really help giving geographic Locations in the World some significance, especially if some of them are super rare.

Also is Coltan Dust supposed to have the [Asbestos] Tooltip? Pretty sure Columbite and Tantalite are not in relation to Asbestos, unlike stuff such as Jade is.

**Comment by HbmMods** [comment]
> Wait how the heck would THAT happen?!

Never underestimate the amount of wacky stuff mods can do :p

Bedrock coltan spawns between Y:1 and 4 which is in line with my most recent tests. Not sure how that would end up at around Y:30.

One block of bedrock ore is sufficient for my system since the mining drill can only extract resources from the first found block within range, adding more wouldn't make that much of a difference except look a bit better visually.

Crushed coltan got the [Asbestos] trait in order to make handling (in addition to finding, processing and using) it even more of a nightmare which is kind of the entire theme around 528 mode. Since I've recently finished the improved item hazard system (but haven't applied it to most items yet) it will most likely get its own poison trait.

**Comment by GregoriusT** [comment]
> One block of bedrock ore is sufficient for my system since the mining drill can only extract resources from the first found block within range, adding more wouldn't make that much of a difference except look a bit better visually.

That sounds like you could plaster a hundred drills around a single Bedrock spot.

As for your Bedrock Ore, I only saw an "x, y, z" in the code, I did not see where the Y was assigned at that moment.

And if it generates between Y1 and Y4, doesn't that mean it might cause issues with Flat Bedrock Modes of other Mods? Since you didn't say Y0.

There is also other Mods ways of crushing Coltan Ore (only GT6, to be fair), so make sure it is not just "dustColtan" that has the Hazard. Like just check for OreDict Strings that contain the word "Coltan" in it, since it is a rather unique word, of which I dont think the substring exists anywhere else. At least that would be the easy way out. Unless there is a good way for me to be compatible with your Hazard System.

**Comment by HbmMods** [comment]
Drills have a relatively small range so the abusability is relatively low.

Bedrock ore gen can be solved easily by simply going from Y:4 to 0, top to bottom, and replacing the first found bedrock with the ore, that way the ore would always be accessible from the top while also respecting whatever the bedrock gen is at the time.

The new hazard system doesn't have a substring functionality, but I'll think about how to solve this once I'm done migrating all the current hazard entries. I'm also planning on adding IMC functionality so other mods can add their own hazard entries or blacklist items to keep them out of the system entirely.

**Comment by GregoriusT** [comment]
One thing I would like to mention, so there is no doubled up work, because September is busy for me right now, and I only do significant Dev things on Weekends.

Since GT6 adds Stone Layers similar to Underground Biomes, clearly there needs to be some way (on my own end) to turn the background layer of Coltan Ore to whatever Stone it resides in.
The Solution I will definitely go with is, that I will generate GT6 Tantalite/Columbite/Coltan Ore myself in that radius, when your Mod is installed, and disable my normally spawning Tantalite/Columbite/Coltan Veins and Stuffs.
Additionally I will likely put one of my own Bedrock Ore Veins very close to wherever the Coltass points to, in order to make sure there is a good "contestable" single point Source for that Stuff.

I presume the Seed based Coltass Pointing Location will never change in the future, so I can just copypaste the RNG for its Location?
```
Random colRand = new Random(world.getSeed() + 5);
int colX = (int) (colRand.nextGaussian() * 1500);
int colZ = (int) (colRand.nextGaussian() * 1500);
```

I will eventually™ also make Tantalum harder to process in GT6 too, since there is only high tier Stuff that really uses Tantalum, such as Tantalum-Hafnium-Carbide, which is an Alloy that has a higher Melting Point than even Carbon or Tungsten.
Funny thing about that Alloy is, I gave it the Color Green, because there is not many Green Metals, and since I personally shortened the name to "TaHfC" I did not notice that it can also be shortened to "THC" until someone told me. XD

tl;dr

You don't need to do ANYTHING regarding this, I just wanted to inform you of what I want to do. ;D

**Comment by HbmMods** [comment]
The random function for the coltan deposit location will never change, however in the future I might add a config for the deposit's diameter, although those will usually stay within the same magnitude in size so I wouldn't really worry about it.
